src/providers/ogr/qgsgeopackagedataitems.cpp

```diff
@@ -370,22 +369,39 @@ QList<QAction *> QgsGeoPackageVectorLayerItem::actions()
 
 void QgsGeoPackageVectorLayerItem::deleteLayer()
 {
-  if ( QMessageBox::question( nullptr, QObject::tr( "Delete Layer" ),
-                              QObject::tr( "Are you sure you want to delete layer '%1' from GeoPackage?" ).arg( mName ),
-                              QMessageBox::Yes | QMessageBox::No, QMessageBox::No ) != QMessageBox::Yes )
-    return;
-
-  QString errCause;
-  bool res = ::deleteLayer( mUri, errCause );
-  if ( !res )
+  // Check if the layer is in the registry
+  const QgsMapLayer *projectLayer = nullptr;
+  Q_FOREACH ( const QgsMapLayer *layer, QgsProject::instance()->mapLayers() )
```

Comment:
Sorry, I'm just asking a side question here about the Q_FOREACH, I thought it was deprecated by QT, especially in QT 5.9. Are we still introducing new Q_FOREACH in the code?
Source: https://www.kdab.com/goodbye-q_foreach/